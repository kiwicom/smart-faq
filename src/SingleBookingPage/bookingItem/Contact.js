// @flow

import * as React from 'react';
import idx from 'idx';
import { TextLink, Typography } from '@kiwicom/orbit-components';
import { Chat, ChevronDown, Phone } from '@kiwicom/orbit-components/lib/icons';
import { createFragmentContainer, graphql } from 'react-relay';

import type { Contact_booking } from './__generated__/Contact_booking.graphql';

type Props = {
  booking: Contact_booking,
};

const Contact = (props: Props) => {
  const { booking } = props;
  const phone = idx(booking, _ => _.contactDetails.phone) || '';
  const email = idx(booking, _ => _.contactDetails.email) || '';
  return (
    <div className="contact">
      <hr />
      <div className="title">
        <Typography size="large" type="attention">
          Contacts
        </Typography>
      </div>
      <div className="contact-options">
        {phone ? (
          <div className="phone">
            <div className="phone-icon inline-icon">
              <Phone customColor="#00a991" />
            </div>
            <Typography size="normal" type="active">
              {phone}
            </Typography>
            <div className="chevron-icon inline-icon">
              <ChevronDown size="small" customColor="#00a991" />
            </div>
          </div>
        ) : null}
        {email ? (
          <div className="chat">
            <div className="chat-icon inline-icon">
              <Chat customColor="#00a991" />
            </div>
            <TextLink
              onClick={() => {}}
              url={`mailto:${email}`}
              title="Write us a message"
            >
              Write us a message
            </TextLink>
          </div>
        ) : null}
      </div>
      <div className="instructions">
        <ol>
          <li>
            When contacting us we will ask the <b>full name</b> of at least one
            of the passengers.
          </li>
          <li>
            The full <b>email address</b> provided during the booking.
          </li>
        </ol>
      </div>
      <style jsx>
        {`
          .contact {
            margin-top: 32px;
          }
          .contact .title {
            margin-bottom: 18px;
          }
          .contact-options,
          .phone,
          .chat,
          .inline-icon {
            display: flex;
            align-items: center;
          }
          .phone {
            margin-right: 40px;
          }
          hr {
            margin-bottom: 16px;
            border: none;
            border-bottom: 1px solid #e8edf1;
          }
          .phone-icon {
            margin-right: 7px;
          }
          .chevron-icon {
            margin-left: 5px;
          }
          .chat-icon {
            margin-right: 14px;
          }
          .instructions {
            margin-left: 20px;
            margin-top: 34px;
            font-size: 14px;
            color: #46515e;
          }
          .instructions ol {
            list-style-type: decimal;
          }
          li {
            margin-bottom: 7px;
            padding-left: 10px;
          }
        `}
      </style>
    </div>
  );
};

export const RawContact = Contact;

export default createFragmentContainer(
  Contact,
  graphql`
    fragment Contact_booking on BookingInterface {
      contactDetails {
        phone
        email
      }
    }
  `,
);
