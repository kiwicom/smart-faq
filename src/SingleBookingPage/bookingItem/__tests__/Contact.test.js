// @flow

import * as React from 'react';
import { shallow, render } from 'enzyme';

import { RawContact } from '../Contact';

describe('Contact', () => {
  const booking = {
    contactDetails: {
      phone: '12345678',
      email: 'kiwi@email.com',
    },
  };
  it('should render with email and phone', () => {
    expect(render(<RawContact booking={booking} />)).toMatchSnapshot();
  });

  it('should render only email', () => {
    const booking = {
      contactDetails: {
        phone: '',
        email: 'kiwi@email.com',
      },
    };
    expect(shallow(<RawContact booking={booking} />)).toMatchSnapshot();
  });
  it('should not render email and phone', () => {
    const booking = {
      contactDetails: {
        phone: '',
        email: '',
      },
    };
    expect(shallow(<RawContact booking={booking} />)).toMatchSnapshot();
  });
});
