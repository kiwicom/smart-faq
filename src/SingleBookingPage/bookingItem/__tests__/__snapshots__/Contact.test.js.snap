// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact should render only email 1`] = `
<div
  className="jsx-2410063808 contact"
>
  <hr
    className="jsx-2410063808"
  />
  <div
    className="jsx-2410063808 contact-options"
  >
    <ContactPageLink
      available={true}
      text="Call us"
      textColor="#00a991"
    />
  </div>
  <div
    className="jsx-2410063808 instructions"
  >
    <ol
      className="jsx-2410063808"
    >
      <li
        className="jsx-2410063808"
      >
        When contacting us we will ask the 
        <b
          className="jsx-2410063808"
        >
          full name
        </b>
         of at least one of the passengers.
      </li>
      <li
        className="jsx-2410063808"
      >
        The full 
        <b
          className="jsx-2410063808"
        >
          email address
        </b>
         provided during the booking.
      </li>
    </ol>
  </div>
  <JSXStyle
    css=".contact.jsx-2410063808{margin-top:32px;}.contact.jsx-2410063808 .title.jsx-2410063808{margin-bottom:18px;}.contact-options.jsx-2410063808,.phone.jsx-2410063808,.chat.jsx-2410063808,.inline-icon.jsx-2410063808{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}.phone.jsx-2410063808{margin-right:40px;}hr.jsx-2410063808{margin-bottom:16px;border:none;border-bottom:1px solid #e8edf1;}.phone-icon.jsx-2410063808{margin-right:7px;}.chevron-icon.jsx-2410063808{margin-left:5px;}.chat-icon.jsx-2410063808{margin-right:14px;}.instructions.jsx-2410063808{margin-left:20px;margin-top:34px;font-size:14px;color:#46515e;}.instructions.jsx-2410063808 ol.jsx-2410063808{list-style-type:decimal;}li.jsx-2410063808{margin-bottom:7px;padding-left:10px;}"
    styleId="2410063808"
  />
</div>
`;

exports[`Contact should render with email and phone 1`] = `
<div
  class="jsx-2410063808 contact"
>
  <hr
    class="jsx-2410063808"
  />
  <div
    class="jsx-2410063808 contact-options"
  >
    <div
      class="jsx-4293162489 contactUs"
    >
      <a
        class="jsx-4293162489"
        href="https://www.kiwi.com/en/content/feedback"
        rel="noopener noreferrer"
        target="_blank"
      >
        Call us
        <span
          class="jsx-4293162489 open-icon"
        >
          <svg
            fill="currentColor"
            height="16px"
            preserveAspectRatio="xMidYMid meet"
            style="vertical-align:middle;color:#00a991"
            viewBox="0 0 24 24"
            width="16px"
          >
            <path
              d="M19 19H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
          </svg>
        </span>
      </a>
    </div>
  </div>
  <div
    class="jsx-2410063808 instructions"
  >
    <ol
      class="jsx-2410063808"
    >
      <li
        class="jsx-2410063808"
      >
        When contacting us we will ask the 
        <b
          class="jsx-2410063808"
        >
          full name
        </b>
         of at least one of the passengers.
      </li>
      <li
        class="jsx-2410063808"
      >
        The full 
        <b
          class="jsx-2410063808"
        >
          email address
        </b>
         provided during the booking.
      </li>
    </ol>
  </div>
</div>
`;
